custom/enikshay/reports/views.py

```diff
@@ -0,0 +1,37 @@
+from django.http.response import JsonResponse
+from django.utils.decorators import method_decorator
+from django.views.generic.base import View
+
+from corehq.apps.domain.decorators import login_and_domain_required
+from corehq.apps.locations.permissions import location_safe
+from corehq.apps.userreports.reports.filters.choice_providers import ChoiceQueryContext, LocationChoiceProvider
+from custom.enikshay.reports.utils import Report
+
+
+class LocationsView(View):
+
+    @method_decorator(login_and_domain_required)
+    @method_decorator(location_safe)
```

Comment:
Looks like you just merged the other PR, so @kkrampa can pull in master and use that version.  I haven't tested this wrapped with `@method_decorator`, but it should work out of the box as a method decorator on a dispatch method or as a class decorator on a django view.

My preference would be:
```python
@location_safe
class LocationsView(View):
    ...
```
As is now done for the other class-based views.